src/Popup.tsx

```diff
@@ -136,6 +143,9 @@ class Popup extends Component<PopupProps, PopupState> {
     const { status } = this.state;
     const { getRootDomNode, visible, stretch } = this.props;
 
+    // If there is a pending state update, cancel it, a new one will be set if necessary
+    this.cancelFrameState();
```

Comment:
This will block status go next if parent node re-render.

Reply:
But if this happens then the code after this will call `setStateOnNextFrame` again if necessary, so status will still be changed.

Without this, then if `visible` is changed to false *before* the next animation frame happens, then when the frame does happen it will change the status away from `stable` and cause the popup to appear when it shouldn't.